src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -567,17 +568,27 @@ private void DoStart()
             // Initialize storage providers once we have a basic silo runtime environment operating
             storageProviderManager = new StorageProviderManager(grainFactory, Services, siloProviderRuntime);
             scheduler.QueueTask(
-                () => storageProviderManager.LoadStorageProviders(GlobalConfig.ProviderConfigurations),
+                () => storageProviderManager.LoadProviders(GlobalConfig.ProviderConfigurations),
                 providerManagerSystemTarget.SchedulingContext)
                     .WaitWithThrow(initTimeout);
             catalog.SetStorageManager(storageProviderManager);
             allSiloProviders.AddRange(storageProviderManager.GetProviders());
             if (logger.IsVerbose) { logger.Verbose("Storage provider manager created successfully."); }
 
+            // Initialize event-storage providers once we have a basic silo runtime environment operating
+            eventStorageProviderManager = new EventStorageProviderManager(grainFactory, Services, siloProviderRuntime);
```

Comment:
It seems like this should be registered with and resolved from the container as `IEventStorageProviderManager`. That will make it easier for us to isolate this into its own library when improve service lifecyle support and split our libraries up more.